gptel-openai.el

```diff
@@ -73,16 +74,24 @@
     (apply #'concat (nreverse content-strs))))
 
 (cl-defmethod gptel--parse-response ((_backend gptel-openai) response _info)
-  (map-nested-elt response '(:choices 0 :message :content)))
+  ;; If the reply specifies a function call, parse and return it instead of the message
```

Comment:
If GPT decides a function should be called, the structured data is included in `tool_calls` instead of `content` so that parsing needs to be updated.
For now I'm only parsing the full response, streaming is not supported yet (but could be)